### PR TITLE
acfl: add post-installation check by running examples

### PR DESCRIPTION
**Pull request by annop-w**
Check installation by running examples and compare results
with expected ones. The makefile is included as part of the
package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/acfl/package.py [hunk 1]
```diff
@@ -131,6 +131,8 @@ class Acfl(Package):
     # Build Versions: establish OS for URL
     acfl_os = get_os()
 
+    test_requires_compiler = True
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/acfl/package.py]: This property is only used for stand-alone, post-installation tests that are run days to weeks after the installation (see https://spack.readthedocs.io/en/latest/packaging_guide.html#enabling-test-compilation).  

What you are adding here is an always on, build-time check (see https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-installation-phase-tests).

File: var/spack/repos/builtin/packages/acfl/package.py [hunk 2]
```diff
@@ -276,3 +278,12 @@ def setup_run_environment(self, env):
         env.prepend_path("PATH", join_path(arm_dir, "bin"))
         env.prepend_path("CPATH", join_path(arm_dir, "include"))
         env.prepend_path("MANPATH", join_path(arm_dir, "share", "man"))
+
+    @run_after("install")
+    def check_install(self):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/acfl/package.py:281]: If you want this to only run when passing `--test` option (see https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-installation-phase-test)), you'll want to add the package attributes directive before the method definition.  There is an example after the table at the link above.

```suggestion
    @on_package_attributes(run_tests=True)
    def check_install(self):
```

Comment (annop-w) [reply var/spack/repos/builtin/packages/acfl/package.py:281]: I had tried adding `@on_package_attributes(run_tests=True)`, but it does not work when passing `--test` option. So, for now I would like this to always run post-installation.

File: var/spack/repos/builtin/packages/acfl/package.py (hunk 1 above)
Comment (annop-w) [reply var/spack/repos/builtin/packages/acfl/package.py]: This was left here by mistake from when I was experimenting. I am going to remove this line.

File: var/spack/repos/builtin/packages/acfl/package.py (hunk 2 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/acfl/package.py:281]: Did you pass it as `--test=root`?

Comment (annop-w) [reply var/spack/repos/builtin/packages/acfl/package.py:281]: Yes. I observed the same behavior with some other packages as well. Perhaps I can try to reproduce it again and raise an issue.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/acfl/package.py:281]: Up to you.  If you want the tests to run after every install and, ideally, they run quickly, then we could leave this as-is.

You could also consider switching the checks to be stand-alone, post-install checks as described https://spack.readthedocs.io/en/latest/packaging_guide.html#stand-alone-or-smoke-tests.  Then they would only be performed -- against the installed software -- when someone runs e.g. `spack test run acfl`.

Comment (annop-w) [reply var/spack/repos/builtin/packages/acfl/package.py:281]: These tests run rather quickly. So, I would like to leave it as-is. I will find time to try to reproduce the problem I saw with `@on_package_attributes(run_tests=True)` and raise an issue, if I can reproduce. 

I had considered making the stand-alone check as well (hence the leftover line above), but that is perhaps for another commit. For now, I will just push a new commit to remove the line above.

File: var/spack/repos/builtin/packages/acfl/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/acfl/package.py]: Ping me when this is removed.

Comment (annop-w) [reply var/spack/repos/builtin/packages/acfl/package.py]: Done.

**Review by tldahlgren — APPROVED** [review]
I was not able to install and test on my local machine but LGTM.

**Pull request merged**